packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import uuid from "uuid";
+import { CreateContainerError } from "@fluidframework/container-utils";
+import { IDocumentStorageService } from "@fluidframework/driver-definitions";
+import { canRetryOnError, DocumentStorageServiceProxy } from "@fluidframework/driver-utils";
+import { ISnapshotTree, IVersion } from "@fluidframework/protocol-definitions";
+import { DeltaManager, getRetryDelayFromError } from "./deltaManager";
+
+export class RetriableDocumentStorageService extends DocumentStorageServiceProxy {
+    private disposed = false;
+    constructor(
+        internalStorageService: IDocumentStorageService,
+        private readonly deltaManager: Pick<DeltaManager, "emitDelayInfo" | "cancelDelayInfo">,
```

Comment:
I think it would be cleaner if we defined interface with these two methods (or at least a type using same Pick<> thing) in DeltaManager itself (or above) and used here and in DeltaManager.

Reply:
We are using this only here. So I don't think we should create a type or add these to interface. We can do that if in future we need to use this somewhere else too.